File: programs/src/plist.c
#include <stdio.h>
#include <utils.h>
int main(size_t argc, OsStr **argv) {
  SysInfo info = {};
  sysinfo(&info);
  ProcessInfo processes[info.processes_count];

  pcollect(processes, info.processes_count);

  printf("name:  pid  ppid\n");
  for (int i = 0; i < info.processes_count; i++) {
    ProcessInfo process = processes[i];

    printf("%s:  %d  %d\n", process.name, process.pid, process.ppid);
  }
  return 0;
}

File: bin/src/plist.zig
const libc = @import("libc");
const sysinfo = libc.sys.utils.zsysinfo;
const zalloc = libc.stdlib.zalloc;
const pcollect = libc.sys.utils.zpcollect;
const printf = libc.stdio.zprintf;

pub fn main() !void {
    const info = try sysinfo();
    const processes = zalloc(libc.sys.raw.ProcessInfo, info.processes_count).?;

    _ = try pcollect(processes);

    try printf("name:  pid  ppid\n", .{});
    for (processes) |process| {
        try printf("\x1B[38;2;0;255;0m%s\x1B[0m:  %d  %d\n", .{ &process.name, process.pid, process.ppid });
    }
}

comptime {
    _ = libc;
}
